src/nodewatcher/nodewatcher.py

```diff
@@ -278,6 +284,22 @@ def _init_idletime():
     return idletime
 
 
+def _lock_and_terminate(scheduler_module, hostname, asg_name, asg_conn, instance_id):
+    _lock_host(scheduler_module, hostname)
+    has_jobs = _has_jobs(scheduler_module, hostname)
```

Comment:
minor: we could avoid the `has_jobs` variable since it is used in a single place and the name of the function called is autoexplicative.